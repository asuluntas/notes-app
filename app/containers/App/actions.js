import {
  LOAD_NOTES,
  LOAD_NOTES_SUCCESS,
  LOAD_NOTES_ERROR,
} from '../Notes/constants';

import {
  ADD_NOTE,
  ADD_NOTE_SUCCESS,
  ADD_NOTE_ERROR,
} from '../HomePage/constants';

export function loadNotes() {
  return {
    type: LOAD_NOTES,
  };
}

export function notesLoaded(notes, note) {
  return {
    type: LOAD_NOTES_SUCCESS,
    notes,
    note,
  };
}

export function notesLoadingError(error) {
  return {
    type: LOAD_NOTES_ERROR,
    error,
  };
}

export function addNote(note) {
  return {
    type: ADD_NOTE,
    note,
  };
}

export function noteAdded(note) {
  return {
    type: ADD_NOTE_SUCCESS,
    note,
  };
}

export function noteAddingError(error) {
  return {
    type: ADD_NOTE_ERROR,
    error,
  };
}
