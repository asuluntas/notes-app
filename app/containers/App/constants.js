export const LOAD_NOTES = 'boilerplate/App/LOAD_NOTES';
export const LOAD_NOTES_SUCCESS = 'boilerplate/App/LOAD_NOTES_SUCCESS';
export const LOAD_NOTES_ERROR = 'boilerplate/App/LOAD_NOTES_ERROR';
export const ADD_NOTE = 'boilerplate/App/ADD_NOTE';
export const ADD_NOTE_SUCCESS = 'boilerplate/App/ADD_NOTE_SUCCESS';
export const ADD_NOTE_ERROR = 'boilerplate/App/ADD_NOTE_ERROR';
