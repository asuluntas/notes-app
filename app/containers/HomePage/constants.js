export const CHANGE_NOTE = 'boilerplate/Home/CHANGE_NOTE';
export const ADD_NOTE = 'boilerplate/Home/ADD_NOTE';
export const ADD_NOTE_SUCCESS = 'boilerplate/Home/ADD_NOTE_SUCCESS';
export const ADD_NOTE_ERROR = 'boilerplate/Home/ADD_NOTE_ERROR';
