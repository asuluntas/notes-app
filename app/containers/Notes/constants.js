export const LOAD_NOTES = 'boilerplate/Notes/LOAD_NOTES';
export const LOAD_NOTES_SUCCESS = 'boilerplate/Notes/LOAD_NOTES_SUCCESS';
export const LOAD_NOTES_ERROR = 'boilerplate/Notes/LOAD_NOTES_ERROR';
