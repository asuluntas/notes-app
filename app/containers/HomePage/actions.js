import { CHANGE_NOTE } from './constants';

export function changeNote(note) {
  return {
    type: CHANGE_NOTE,
    note,
  };
}
