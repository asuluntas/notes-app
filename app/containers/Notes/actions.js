import { LOAD_NOTES, LOAD_NOTES_SUCCESS, LOAD_NOTES_ERROR } from './constants';

export function loadNotes() {
  return {
    type: LOAD_NOTES,
  };
}

export function notesLoaded(notes) {
  return {
    type: LOAD_NOTES_SUCCESS,
    notes,
  };
}

export function notesLoadingError(error) {
  return {
    type: LOAD_NOTES_ERROR,
    error,
  };
}
