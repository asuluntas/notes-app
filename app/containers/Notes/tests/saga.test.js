import { put, takeLatest } from 'redux-saga/effects';

import { LOAD_NOTES } from '../constants';
import { notesLoaded, notesLoadingError } from '../actions';

import watchLoadNotes, { getNotes } from '../saga';

const note = 'a test note';

/* eslint-disable redux-saga/yield-effects */
describe('getNotes Saga', () => {
  let getNotesGenerator;

  // We have to test twice, once for a successful load and once for an unsuccessful one
  // so we do all the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    getNotesGenerator = getNotes();

    const selectDescriptor = getNotesGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    const callDescriptor = getNotesGenerator.next(note).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the notesLoaded action if it requests the data successfully', () => {
    const response = [
      {
        id: 3,
        text: 'a test note1',
      },
      {
        id: 4,
        text: 'a test note2',
      },
    ];
    const putDescriptor = getNotesGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(notesLoaded(response)));
  });

  it('should call the notesLoadingError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = getNotesGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(notesLoadingError(response)));
  });
});

describe('watchLoadNotesSaga Saga', () => {
  const watchLoadNotesSaga = watchLoadNotes();

  it('should start task to watch for LOAD_NOTES action', () => {
    const takeLatestDescriptor = watchLoadNotesSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(LOAD_NOTES, getNotes));
  });
});
