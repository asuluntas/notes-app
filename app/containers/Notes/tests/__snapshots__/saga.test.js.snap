// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getNotes Saga should call the notesLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/fetchNotes",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getNotes Saga should call the notesLoadingError action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "notes": "a test note",
      "type": "boilerplate/App/LOAD_NOTES_SUCCESS",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`getNotes Saga should dispatch the notesLoaded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/fetchNotes",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getNotes Saga should dispatch the notesLoaded action if it requests the data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "notes": "a test note",
      "type": "boilerplate/App/LOAD_NOTES_SUCCESS",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;
